### PR TITLE
Wip dang zipper raw obj

### PR DESCRIPTION
**Pull request by dang**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/rgw/rgw_sal.h [hunk 1]
```diff
@@ -378,6 +378,10 @@ class Bucket {
     virtual Attrs& get_attrs(void) { return attrs; }
     virtual int set_attrs(Attrs a) { attrs = a; return 0; }
     virtual int remove_bucket(const DoutPrefixProvider* dpp, bool delete_children, std::string prefix, std::string delimiter, bool forward_to_master, req_info* req_info, optional_yield y) = 0;
+    virtual int remove_bucket_bypass_gc(int concurrent_max, bool
```
Comment (mattbenjamin) [inline src/rgw/rgw_sal.h:385]: Is this a permanent api method?  I would have said that gc is an RGWRados internal

Comment (dang) [reply src/rgw/rgw_sal.h:385]: Probably not.  It's used by admin ops currently.

**Comment by dang** [comment]
jenkins test make check

**Comment by dang** [comment]
jenkins test make check arm64

**Comment by dang** [comment]
https://pulpito.ceph.com/dang-2021-07-28_11:34:53-rgw-wip-dang-zipper-raw_obj-distro-basic-smithi/

**Comment by dang** [comment]
@cbodley @soumyakoduri This is ready to merge, if it can be reviewed.

**Review by soumyakoduri — APPROVED** [review]

**Pull request merged**